Makefile

```diff
@@ -34,3 +34,110 @@ binary:
 	@echo "Making $@ ..."
 	@docker pull neuvector/build_fleet:${BUILD_IMAGE_TAG}
 	@docker run --rm -ia STDOUT --name build -e VULN_VER=$(VULN_VER) --net=none -v $(CURDIR):/go/src/github.com/neuvector/scanner -w /go/src/github.com/neuvector/scanner --entrypoint ./make_bin.sh neuvector/build_fleet:${BUILD_IMAGE_TAG}
+
+RUNNER := docker
+IMAGE_BUILDER := $(RUNNER) buildx
+MACHINE := neuvector
+BUILDX_ARGS ?= --sbom=true --attest type=provenance,mode=max
+DEFAULT_PLATFORMS := linux/amd64,linux/arm64,linux/x390s,linux/riscv64
+STAGE_DIR=stage
+
+# For scanner, the version is also vulndb version.
+COMMIT = $(shell git rev-parse --short HEAD)
+ifeq ($(VERSION),)
+	# Define VERSION, which is used for image tags or to bake it into the
+	# compiled binary to enable the printing of the application version, 
+	# via the --version flag.
+	CHANGES = $(shell git status --porcelain --untracked-files=no)
+	ifneq ($(CHANGES),)
+		DIRTY = -dirty
+	endif
+
+	GIT_TAG = $(shell git tag -l --contains HEAD | head -n 1)
+
+	COMMIT = $(shell git rev-parse --short HEAD)
+	VERSION = $(COMMIT)$(DIRTY)
+
+	# Override VERSION with the Git tag if the current HEAD has a tag pointing to
+	# it AND the worktree isn't dirty.
+	ifneq ($(GIT_TAG),)
+		ifeq ($(DIRTY),)
+			VERSION = $(GIT_TAG)
+		endif
+	endif
+	VULNDBVER=LATEST
+else
+	VULNDBVER=$(VERSION:1)
+endif
+
+ifeq ($(TAG),)
+	TAG = $(VERSION)
+	ifneq ($(DIRTY),)
+		TAG = dev
+	endif
+endif
+
+TARGET_PLATFORMS ?= linux/amd64,linux/arm64
+REPO ?= neuvector
+IMAGE = $(REPO)/scanner:$(TAG)
+BUILD_ACTION = --load
+
+.PHONY: all copy_scan build
+
+ARCH := $(shell uname -p)
+
+BASE_IMAGE_TAG = latest
+BUILD_IMAGE_TAG = v2
+
+# Keep this as the first
+slsa_all: test build copy_scan_slsa
+
+build:
+	go build -ldflags='-s -w' -buildvcs=false
+	make -C task/
+	make -C monitor/
+
+test:
+	# Only run unit-test on amd64 for now
+	if [ "$(ARCH)" = "x86_64" ]; then go test ./...;fi
```

Comment:
Is there an issue with the testing on other supported platforms?

Reply:
No.  It's just very slow.  We will revisit when we have the first batch of release.

Reply:
> It's just very slow when running in emulated ARM64 architecture for a release

The caller should be responsible for deciding whether or not to call `make test`, instead of making this decision here.
Let's revisit this on a follow-up PR.